File: days/day02/main.cpp
#include "aoclib.hpp"
#include <vector>

int solve(std::string_view input) {
    int points{0};
    const auto them{input[0]};
    const auto us{input[2]};
    if (us == 'X') {
        points += 1;
        if (them == 'A') {
            points += 3;
        } else if (them == 'B') {
            // Lose no points.
        } else if (them == 'C') {
            points += 6;
        }
    } else if (us == 'Y') {
        points += 2;
        if (them == 'A') {
            points += 6;
        } else if (them == 'B') {
            points += 3;
        } else if (them == 'C') {
            // Lose no points.
        }
    } else if (us == 'Z') {
        points += 3;
        if (them == 'A') {
            // Lose no points.
        } else if (them == 'B') {
            points += 6;
        } else if (them == 'C') {
            points += 3;
        }
    }
    return points;
}

int main(int argc, const char* argv[]) {
    const auto part{parse_args(argc, argv)};
    std::ifstream file{"days/day02/input"};
	const std::vector<std::string> file_data{IfstreamIterator{file}, {}};   
    int result{};
    if (part == 1) {
        for(const auto& line: file_data) {
            result += solve(line);
        }
        std::cout << result;
    } else if (part == 2) {
    }
}

File: days/day02/main.rs
/// returns points for the round.
fn solve(input: &str) -> i32 {
    // input example: "A X"
    let mut points = 0;
    let (them, us) = input.split_once(' ').unwrap();
    if us == "X" {
        points += 1;
        points += match them {
            "A" => 3,
            "B" => 0,
            "C" => 6,
            _ => 0,
        }
    } else if us == "Y" {
        points += 2;
        points += match them {
            "A" => 6,
            "B" => 3,
            "C" => 0,
            _ => 0,
        }
    } else if us == "Z" {
        points += 3;
        points += match them {
            "A" => 0,
            "B" => 6,
            "C" => 3,
            _ => 0,
        }
    }
    points
}
fn solve2(input: &str) -> i32 {
    // input example: "A X"
    let mut points = 0;
    let (them, result) = input.split_once(' ').unwrap();
    if result == "X" {
        points += match them {
            "A" => 3,
            "B" => 1,
            "C" => 2,
            _ => 0,
        }
    } else if result == "Y" {
        points += match them {
            "A" => 4,
            "B" => 5,
            "C" => 6,
            _ => 0,
        }
    } else if result == "Z" {
        points += match them {
            "A" => 8,
            "B" => 9,
            "C" => 7,
            _ => 0,
        }
    }
    points
}

fn main() {
    // A/X rock(1) B/Y paper(2) C/Z scissors(3)
    // 0 lose 3 draw 6 win
    // part2:
    // X lose y draw z win
    let file_data = include_str!("input");
    let part1: i32 = file_data.lines().map(solve).sum();
    println!("{}", part1);
    let part2: i32 = file_data.lines().map(solve2).sum();
    println!("{}", part2);
}
